web/client/epics/usersession.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import Rx from 'rxjs';
+import { error } from '../actions/notifications';
+import { SAVE_USER_SESSION, userSessionSaved, loading, saveUserSession } from "../actions/usersession";
+import { createResource, createCategory, updateResource } from '../api/persistence';
+import {userSelector} from '../selectors/security';
+import { wrapStartStop } from '../observables/epics';
+import isString from "lodash/isString";
+
+const saveUserSessionErrorStatusToMessage = (status) => {
+    switch (status) {
+    default:
+        return 'userSession.saveErrorNotification.defaultMessage';
+    }
+};
+
+/**
+ * Returns a user session save epic.
+ * The epic triggers on a SAVE_USER_SESSION action.
+ *
+ * @param {function} nameSelector selector that builds the session identifier
+ * @param {*} sessionSelector selector that builds the session object to save
+ * @param {*} idSelecotr selector of the identifier for the current session (if it already exists)
```

Comment:
```suggestion
 * @param {*} idSelector selector of the identifier for the current session (if it already exists)
```
Is not clear to me the role of the session name and the role of the session id. Can you clarify in JSDoc (maybe the right place is the reducer)